### PR TITLE
Hotfix game crash on load

### PR DESCRIPTION
**Pull request by krzychu124**
Release 11.7.1.2

- temporary hotfix for crash on load

### CODE REVIEW

File: TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs [hunk 1]
```diff
@@ -104,7 +105,8 @@ public abstract class SerializableUIOptionBase<TVal, TUI, TComponent> : ILegacyS
 
         /* UI: */
 
-        public bool HasUI => _ui != null;
+        // TODO FIX - temporary solution!
+        public bool HasUI => _ui != null && Thread.CurrentThread != SimulationManager.instance.m_simulationThread;
```
Comment (kianzarrin) [inline TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs:109]: can you make sure its in main thread?

Comment (krzychu124) [reply TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs:109]: How to get main thread handle?

Comment (kianzarrin) [reply TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs:109]:
```cs
bool InMainThread() => Dispatcher.currentSafe == ThreadHelper.dispatcher;
```

Comment (kianzarrin) [reply TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs:109]: now that I read the code again I am not sure if that really is the main thread.

Comment (krzychu124) [reply TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs:109]: hmm, I'm pretty sure it is, looking in the code, it's just not so obvious reading code of your proposed helper

**Review by kvakvs — APPROVED** [review]

**Review by chameleon-tbn — APPROVED** [review]
lgtm

**Review by kianzarrin — APPROVED** [review]

**Pull request merged**